blaze-server/src/main/scala/org/http4s/blaze/server/Http1ServerStage.scala

```diff
@@ -303,15 +302,19 @@ private[blaze] class Http1ServerStage[F[_]](
             logger.trace("Request/route requested closing connection.")
             F.delay(closeConnection())
           } else
-            F.delay {
+            F.async_[Unit] { cb =>
               bodyCleanup().onComplete {
                 case s @ Success(_) => // Serve another request
-                  parser.reset()
-                  handleReqRead(s)
+                  cb(Right {
+                    parser.reset()
+                    handleReqRead(s)
+                  })
 
-                case Failure(EOF) => closeConnection()
+                case Failure(EOF) =>
+                  cb(Right(closeConnection()))
 
-                case Failure(t) => fatalError(t, "Failure in body cleanup")
+                case Failure(t) =>
+                  cb(Right(fatalError(t, "Failure in body cleanup")))
               }(trampoline)
```

Comment:
> Doesn't renderResponse run on the executionContext, which in modern times will be the one from Async[F]? I'm not following why we're specifically shifting back on this effect.

@rossabaker I think the initial 'todo' was about the `trampoline` execution context. So, we intended to back to `Async`s execution context as soon as possible from the `trampoline` context.